File: ExampleRecyclerviewModeGrid/app/src/main/java/com/gnm/examplerecyclerviewmodegrid/MainActivity.java
package com.gnm.examplerecyclerviewmodegrid;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private String title = "Mode List";
    private RecyclerView rvHeroes;
    private ArrayList<com.gnm.examplerecyclerviewmodegrid.Hero> list = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setActionBarTitle(title);

        rvHeroes = findViewById(R.id.rv_heroes);
        rvHeroes.setHasFixedSize(true);

        list.addAll(HeroesData.getListData());
        showRecyclerGrid();
    }

    private void showRecyclerGrid() {
        rvHeroes.setLayoutManager(new GridLayoutManager(this, 2));
        com.gnm.examplerecyclerviewmodegrid.GridHeroAdapter gridHeroAdapter = new com.gnm.examplerecyclerviewmodegrid.GridHeroAdapter(list);
        rvHeroes.setAdapter(gridHeroAdapter);

        gridHeroAdapter.setOnItemClickCallback(new com.gnm.examplerecyclerviewmodegrid.GridHeroAdapter.OnItemClickCallback() {
            @Override
            public void onItemClicked(com.gnm.examplerecyclerviewmodegrid.Hero data) {
                showSelectedHero(data);
            }
        });
    }

    private void setActionBarTitle(String title) {
        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle(title);
        }
    }

    private void showSelectedHero(com.gnm.examplerecyclerviewmodegrid.Hero hero) {
        Toast.makeText(this, "Kamu memilih " + hero.getName(), Toast.LENGTH_SHORT).show();
    }
}

File: StartNewActivity/app/src/main/java/ajantha/sample/startnewactivity/MainActivity.kt
package ajantha.sample.startnewactivity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button = findViewById<Button>(R.id.button)
        val editTextName = findViewById<EditText>(R.id.edit_text_name)
        val editTextAge = findViewById<EditText>(R.id.edit_text_age)

        button.setOnClickListener {
            val name = editTextName.text.toString().trim()
            val age = editTextAge.text.toString().trim().toInt()

            val intent = Intent(this, WelcomeActivity::class.java)
            intent.putExtra(EXTRA_NAME, name)
            intent.putExtra(EXTRA_AGE, age)
            startActivity(intent)
        }

    }

    companion object {
        const val EXTRA_NAME = "NAME"
        const val EXTRA_AGE = "AGE"
    }

}